client/state/data-layer/wpcom/read/tags/utils.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+import { map, compact, concat, isObject, isArray } from 'lodash';
+import { decodeEntities } from 'lib/formatting';
+
+/**
+ * Normalize response from the api so whether we get back a single tag or a list of tags
+ * we always pass forward a list
+ * Also transform the api response to be something more calypso-friendly
+ *
+ * @param  {Tag|Tags} apiResponse api response from the tags endpoint
+ * @return {Tags} An object containing all of the normalized tags in the format:
+ *  [
+ *    { id, displayName, url, title, slug },
+ *    ...
+ *  ]
+ */
+export function fromApi( apiResponse ) {
+	if ( ! apiResponse || ( ! apiResponse.tag && ! apiResponse.tags ) ) {
+		if ( process.env.NODE_ENV === 'development' ) {
```

Comment:
the entire block gets stripped as dead code, so its safe in production.
that said, @blowery brought up concerns about throwing errors in dev/untested lack of code in production so i'm going to revert to` console.error` since I think they are equally noticeable